lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -0,0 +1,133 @@
+from http import HTTPStatus
+import os
+import werkzeug.utils
+import typing as t
+from datetime import datetime
+from pathlib import Path
+
+import pytest
+import requests
+
+if t.TYPE_CHECKING:
+    from _typeshed.wsgi import WSGIEnvironment
+    from flask.wrappers import Response
+
+from pbench.server import PbenchServerConfig
+from pbench.server.database.models.datasets import Dataset, DatasetNotFound
+from pbench.server.filetree import FileTree
+
+
+class TestDatasetsAccess:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config, more_datasets, provide_metadata):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            provide_metadata: Dataset metadata fixture
+        """
+
+        def query_api(
+            dataset: str, username: str, expected_status: HTTPStatus, path: str
+        ) -> requests.Response:
+            headers = None
+            try:
+                dataset_id = Dataset.query(name=dataset).resource_id
+            except DatasetNotFound:
+                dataset_id = dataset  # Allow passing deliberately bad value
+            if username:
+                token = self.token(client, server_config, username)
+                headers = {"authorization": f"bearer {token}"}
+            response = client.get(
+                f"{server_config.rest_uri}/datasets/inventory/{dataset_id}/{path}",
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+
+            if username:
+                client.post(
+                    f"{server_config.rest_uri}/logout",
+                    headers={"authorization": f"bearer {token}"},
+                )
```

Comment:
Webb brought this up earlier; it's because Siddardh carried this code forward from the metadata tests, which faced some special requirements due to multiple API calls (trying to leverage the same general helpers) within a single test case. Because the second call ended up trying to log in again and got the duplicate `auth_token` issue, it was "cleaner" (or at least expedient) to log out. Not necessary here or in most other tests; and hopefully won't be necessary at all once we go to Keycloak.